prov/util/src/util_attr.c

```diff
@@ -162,8 +162,19 @@ static int ofi_info_to_core(uint32_t version, const struct fi_provider *prov,
 	const char *core_name;
 	size_t len;
 
-	if (!(*core_hints = fi_allocinfo()))
-		return -FI_ENOMEM;
+	/* If we are here through fi_fabric we must allocate only fabric_attr */
+	if (util_info && util_info->fabric_attr && util_info->fabric_attr->name &&
+	    !util_info->domain_attr) {
```

Comment:
If we came here via fi_fabric(), why are the domain attributes being checked at all and where?

I'm not a fan of this change.  It just feels wrong.  The function is supposed to just convert one fi_info structure into another.  Adding call path specific checks is odd.  Plus there is already a check in the function that sees if the util_info has a domain_attr field.  Adding an assumption above the existence of fabric_attr in the absence of domain_attr will be very difficult to maintain/understand by another developer.